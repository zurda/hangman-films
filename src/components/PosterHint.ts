import styled from '@emotion/styled'



interface IPosterProps {
  imgSrc: string
}

const HintContainer = styled.div<IPosterProps>`
  background-image: url(${props => props.imgSrc});
  height: 50px;
  width: 50px;
  background-position: center right;
`

export default HintContainer
