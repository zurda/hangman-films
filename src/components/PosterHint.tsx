import React, { useState } from 'react'
import styled from '@emotion/styled'
import { RoundButton } from './GameControllers'
import { Text } from '../style/shared'

interface IPosterProps {
  imgSrc: string
}

const HintContainer = styled.div<IPosterProps>`
  background-image: url(${props => props.imgSrc});
  height: 50px;
  width: 50px;
  background-position: center right;
`

interface IHint {
  counter: number
  poster: string
  setCounter: (n: number) => void
}

const PosterHint = ({ counter, poster, setCounter }: IHint) => {
  const [isHinted, setIsHinted] = useState<boolean>(false)

  const onHintClick = () => {
    setCounter(counter - 2)
    setIsHinted(true)
  }

  return (
    <>
      {!isHinted && counter > 2 &&
        (<>
          <RoundButton onClick={onHintClick}>Hint</RoundButton>
          <Text>This will cost you two guesses!</Text>
        </>)}
      {isHinted && counter > 0 &&
        (<HintContainer imgSrc={`http://image.tmdb.org/t/p/w154${poster}`} />)
      }
    </>
  )
}

export default PosterHint