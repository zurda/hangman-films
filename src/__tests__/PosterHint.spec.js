import React from 'react'
import { create } from 'react-test-renderer'
import PosterHint from '../components/PosterHint'

const posterData = 'sM33SANp9z6rXW8Itn7NnG1GOEs.jpg'
const counterData = 5

describe('PosterHint component', () => {
  test('Matches the snapshot', () => {
    const posterHint = create(
      <PosterHint
        poster={posterData}
        setCounter={(n) => null}
        counter={counterData}
      />
    )
    expect(posterHint.toJSON()).toMatchSnapshot()
  })
})
