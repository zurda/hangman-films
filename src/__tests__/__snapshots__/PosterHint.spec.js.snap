// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterHint component Matches the snapshot 1`] = `
Array [
  <button
    className="css-15j8xbs"
    onClick={[Function]}
  >
    Hint
  </button>,
  <p
    className="css-ck8036"
  >
    This will cost you two guesses!
  </p>,
]
`;
